dynawo/sources/Solvers/SolverSIM/DYNSolverEulerKIN.cpp

```diff
@@ -146,13 +161,16 @@ SolverEulerKIN::init(const shared_ptr<Model>& model, const std::string& linearSo
   // (6) Solver choice
   // -------------------
   // Here CSR has nothing to do with how the matrix is stored but rather how to solve the linear system using the matrix (CSC_MAT) or its transpose (CSR_MAT)
+  M_ = SUNSparseMatrix(sizeY, sizeY, 10., CSR_MAT);
   if (linearSolverName_ == "KLU") {
-    flag = KINKLU(KINMem_, sizeY, 10., CSR_MAT);
+    LS_ = SUNLinSol_KLU(yy_, M_);
+    flag = KINSetLinearSolver(KINMem_, LS_, M_);
     if (flag < 0)
       throw DYNError(Error::SUNDIALS_ERROR, SolverFuncErrorKINSOL, "KINKLU");
 #ifdef WITH_NICSLU
   } else if (linearSolverName_ == "NICSLU") {
-    flag = kinNICSLU(KINMem_, sizeY, 10., CSR_MAT);
+    LS_ = SUNLinSol_NICSLU(yy_, M_);
```

Comment:
Again testing if M_ and LS_ are created without any problem?

Reply:
Solved